### PR TITLE
Add helm chart value for image digest

### LINKED ISSUE #161: Allow Chart to Accept Image Digest
Hi @ricoberger 👋 

My org is in the process of rolling out image signing with [Cosign](https://github.com/sigstore/cosign). One of the requirements of this tool is that image SHAs should be used rather than tags. This is so that image signature (which is stored in the registry and tagged with the corresponding image's SHA) can be properly dereferenced.

We also use the vault secrets operator Helm chart. However, it appears that the chart only supports image tags at this time.

I'd like to request that this capability be added to the helm chart.

Thank you!

### PR DESCRIPTION
**Pull request by Zarmbinski**
Closes #161

### CODE REVIEW

File: charts/vault-secrets-operator/templates/_helpers.tpl [hunk 1]
```diff
@@ -95,3 +95,14 @@ Additional containers to add to the deployment
 */}}
 {{- define "vault-secrets-operator.additionalContainers" -}}
 {{- end -}}
+
+{{/*
+Helper function for checking if a property is defined
+*/}}
+{{- define "vault-secrets-operator.imageRef" -}}
+{{- if .Values.image.digest -}}
+    {{ .Values.image.digest }}
```
Comment (ricoberger) [inline charts/vault-secrets-operator/templates/_helpers.tpl:104]:
```suggestion
    @{{ .Values.image.digest }}
```

File: charts/vault-secrets-operator/templates/_helpers.tpl [hunk 2]
```diff
@@ -95,3 +95,14 @@ Additional containers to add to the deployment
...
+{{- else -}}
+    {{ .Values.image.tag }}
```
Comment (ricoberger) [inline charts/vault-secrets-operator/templates/_helpers.tpl:106]:
```suggestion
    :{{ .Values.image.tag }}
```

**Review by ricoberger — CHANGES_REQUESTED** [review]
Hi @Zarmbinski, thanks for your contribution 🙂.

I think you have to add a `:` and `@` in the helpers function.

Can you please also add the `image.digest` property to the `values.yaml` file.

File: charts/vault-secrets-operator/templates/_helpers.tpl (hunk 1 above)
Comment (Zarmbinski) [reply charts/vault-secrets-operator/templates/_helpers.tpl]: Good call. It's been fixed 👍

File: charts/vault-secrets-operator/templates/_helpers.tpl (hunk 2 above)
Comment (Zarmbinski) [reply charts/vault-secrets-operator/templates/_helpers.tpl]: Done

**Comment by Zarmbinski** [comment]
> Hi @Zarmbinski, thanks for your contribution 🙂.
> 
> I think you have to add a `:` and `@` in the helpers function.
> 
> Can you please also add the `image.digest` property to the `values.yaml` file.

Good catch. I've made the requested changes 👍

**Review by ricoberger — APPROVED** [review]

**Comment by ricoberger** [comment]
Thanks again for your work 🙂

**Comment by Zarmbinski** [comment]
> Thanks again for your work 🙂

Thank you! It's an honor to contribute to such an excellent tool

**Pull request merged**